src/vm/methodtablebuilder.cpp

```diff
@@ -1499,7 +1499,18 @@ MethodTableBuilder::BuildMethodTableThrowing(
         LPCUTF8 className;
         LPCUTF8 nameSpace;
         HRESULT hr = GetMDImport()->GetNameOfTypeDef(bmtInternal->pType->GetTypeDefToken(), &className, &nameSpace);
-    
+
+        if (hr == S_OK && strcmp(nameSpace, "System.Runtime.Intrinsics") == 0)
+        {
+            if (IsCompilationProcess())
+            {
+                // Disable AOT compiling for the SIMD hardware intrinsic types. These types require special
+                // ABI handling as they represent fundamental data types (__m64, __m128, and __m256) and not
+                // aggregate or union types. See https://github.com/dotnet/coreclr/issues/15943
+                COMPlusThrow(kTypeLoadException, IDS_EE_HWINTRINSIC_NGEN_DISALLOWED);
```

Comment:
@jkotas @CarolEidt If SIMD types are disabled for ABI, how can we enable HW intrinsics for crossgen (or other AOT platforms) in the future?

Reply:
One we have the right stable ABI for the Vector64/128/256 types, we can re-enable it for crossgen/AOT.

Reply:
@tannergooding Could you please update the comment to capture this?

Reply:
Updated to additionally include the following comment: `Once they are properly handled according to the ABI requirements, we can remove this check and allow them to be used in crossgen/AOT scenarios.`